yb-voyager/src/query/queryissue/detectors_test.go

```diff
@@ -76,9 +76,23 @@ func TestFuncCallDetector(t *testing.T) {
 		`SELECT pg_advisory_unlock_all();`,
 	}
 
-	for _, sql := range advisoryLockSqls {
-		detector := NewFuncCallDetector(sql)
+	loFunctionSqls := []string{
```

Comment:
Let's prefer writing tests in parser_issue_detector. 
We should have basic tests for each detector (i.e. is functionCallDetector able to detect a function call or not). 

But for all issues, and various combinations, better to write them in parserIssueDetector layer because that is the API that is called by assess/analyze. 

Also, as a general note, too many unit tests for the internal structures can sometimes be counter-productive :) . For example, recently I changed the interface of a detector, and then had to go and update all these tests.

Reply:
Agreed, Will only tests here when I will new detector. Rest will be in `parser_issue_Detector_test`